src/dex/lite-psm/lite-psm-event-pool.ts

```diff
@@ -0,0 +1,216 @@
+    );
+    const tout = bigIntify(
+      psmInterface.decodeFunctionResult('tout', res.returnData[i++])[0],
+    );
+    const ilks = vatInterface.decodeFunctionResult('ilks', res.returnData[i++]);
+    const rate = bigIntify(ilks.rate);
+    const daiBalance = bigIntify(
+      erc20Iface.decodeFunctionResult('balanceOf', res.returnData[i++])[0],
+    );
+    const gemBalance = bigIntify(
+      erc20Iface.decodeFunctionResult('balanceOf', res.returnData[i++])[0],
+    );
+    const response = {
+      tin,
+      tout,
+      daiBalance,
+      gemBalance,
+      rate,
+    };
+
+    return response;
+  });
+
+  return result;
+}
+
+export class LitePsmEventPool extends StatefulEventSubscriber<PoolState> {
+  handlers: {
+    [event: string]: (event: any, pool: PoolState, log: Log) => PoolState;
+  } = {};
+
+  logDecoder: (log: Log) => any;
+
+  to18ConversionFactor: bigint;
+  bytes32Tout =
+    '0x746f757400000000000000000000000000000000000000000000000000000000'; // bytes32('tout')
+  bytes32Tin =
+    '0x74696e0000000000000000000000000000000000000000000000000000000000'; // bytes32('tin')
+
+  daiAddress: Address;
+
+  constructor(
+    parentName: string,
+    protected network: number,
+    protected dexHelper: IDexHelper,
+    logger: Logger,
+    public poolConfig: PoolConfig,
+    protected vatAddress: Address,
+  ) {
+    super(parentName, poolConfig.identifier, dexHelper, logger);
+
+    this.daiAddress =
+      LitePsmConfig[this.parentName][this.network].dai.address.toLowerCase();
+    this.logDecoder = (log: Log) => {
+      const logAddress = log.address.toLowerCase();
+      if (logAddress === this.daiAddress) {
+        return daiInterface.parseLog(log);
+      } else if (logAddress === this.poolConfig.gem.address) {
+        return usdcInterface.parseLog(log);
+      } else {
+        return psmInterface.parseLog(log);
+      }
+    };
+    this.addressesSubscribed = [
+      poolConfig.psmAddress,
+      this.daiAddress,
+      this.poolConfig.gem.address,
+    ];
+    this.to18ConversionFactor = getBigIntPow(18 - poolConfig.gem.decimals);
+
+    // Add handlers
+    this.handlers['File'] = this.handleFile.bind(this);
+    this.handlers['Transfer'] = this.handleTransfer.bind(this);
+  }
+
+  handleFile(event: any, pool: PoolState, log: Log): PoolState {
+    if (event.args.what === this.bytes32Tin) {
+      pool.tin = bigIntify(event.args.data);
+    } else if (event.args.what === this.bytes32Tout) {
+      pool.tout = bigIntify(event.args.data);
+    }
+    return pool;
+  }
+
+  handleTransfer(event: any, pool: PoolState, log: Log): PoolState {
+    if (log.address.toLowerCase() === this.daiAddress) {
+      return this.handleTransferDAI(event, pool, log);
+    } else if (log.address.toLowerCase() === this.poolConfig.gem.address) {
+      return this.handleTransferGEM(event, pool, log);
+    }
+    return pool;
+  }
+
+  handleTransferDAI(event: any, pool: PoolState, log: Log): PoolState {
```

Comment:
@JackieJoo The same

Reply:
Because we track 2 different Transfers: Usdc and dai, they have different interfaces and logic in the pool.

Dai transfer has src, dst, wad. 
<img width="736" alt="Screenshot 2024-08-21 at 14 47 58" src="https://github.com/user-attachments/assets/212580b6-2bd3-4015-9a69-1127ee4f984b">

Whereas USDC transfer is erc20 from, to, amount